source/Preferences.cpp

```diff
@@ -118,30 +117,29 @@ namespace {
 		{Preferences::OverlayType::NEUTRAL, Preferences::OverlayState::OFF},
 	};
 
-	const vector<string> AUTO_AIM_SETTINGS = {"off", "always on", "when firing"};
-	int autoAimIndex = 2;
-
-	const vector<string> AUTO_FIRE_SETTINGS = {"off", "on", "guns only", "turrets only"};
-	int autoFireIndex = 0;
-
-	const vector<string> BOARDING_SETTINGS = {"proximity", "value", "mixed"};
-	int boardingIndex = 0;
+	int previousSaveCount = 3;
 
-	const vector<string> FLOTSAM_SETTINGS = {"off", "on", "flagship only", "escorts only"};
-	int flotsamIndex = 1;
+}
 
-	// Enable "fast" parallax by default. "fancy" is too GPU heavy, especially for low-end hardware.
-	const vector<string> PARALLAX_SETTINGS = {"off", "fancy", "fast"};
-	int parallaxIndex = 2;
 
-	const vector<string> EXTENDED_JUMP_EFFECT_SETTINGS = {"off", "medium", "heavy"};
-	int extendedJumpEffectIndex = 0;
 
-	const vector<string> ALERT_INDICATOR_SETTING = {"off", "audio", "visual", "both"};
-	int alertIndicatorIndex = 3;
 
-	int previousSaveCount = 3;
-}
+Preferences::MultiPreference<Preferences::AlertIndicator, 0>Preferences::alertIndicator(
+	{"off", "audio", "visual", "both"});
+Preferences::MultiPreference<Preferences::AutoAim, 2>Preferences::autoAim(
+	{"off", "always on", "when firing"});
+Preferences::MultiPreference<Preferences::AutoFire, 0>Preferences::autoFire(
+	{"off", "on", "guns only", "turrets only"});
+Preferences::MultiPreference<Preferences::BackgroundParallax, 2>Preferences::backgroundParallax(
+	{"off", "fancy", "fast"});
+Preferences::MultiPreference<Preferences::BoardingPriority, 0>Preferences::boardingPriority(
+	{"proximity", "value", "mixed"});
+Preferences::MultiPreference<Preferences::DateFormat, 0>Preferences::dateFormat(
+	{"dd/mm/yyyy", "mm/dd/yyyy", "yyyy-mm-dd"});
+Preferences::MultiPreference<Preferences::ExtendedJumpEffects, 0>Preferences::extendedJumpEffects(
+	{"off", "medium", "heavy"});
+Preferences::MultiPreference<Preferences::FlotsamCollection, 1>Preferences::flotsamCollection(
+	{"off", "on", "flagship only", "escorts only"});
```

Comment:
Feeling a little claustrophobic over here.
```suggestion
Preferences::MultiPreference<Preferences::AlertIndicator, 0> Preferences::alertIndicator(
	{"off", "audio", "visual", "both"});
Preferences::MultiPreference<Preferences::AutoAim, 2> Preferences::autoAim(
	{"off", "always on", "when firing"});
Preferences::MultiPreference<Preferences::AutoFire, 0> Preferences::autoFire(
	{"off", "on", "guns only", "turrets only"});
Preferences::MultiPreference<Preferences::BackgroundParallax, 2> Preferences::backgroundParallax(
	{"off", "fancy", "fast"});
Preferences::MultiPreference<Preferences::BoardingPriority, 0> Preferences::boardingPriority(
	{"proximity", "value", "mixed"});
Preferences::MultiPreference<Preferences::DateFormat, 0> Preferences::dateFormat(
	{"dd/mm/yyyy", "mm/dd/yyyy", "yyyy-mm-dd"});
Preferences::MultiPreference<Preferences::ExtendedJumpEffects, 0> Preferences::extendedJumpEffects(
	{"off", "medium", "heavy"});
Preferences::MultiPreference<Preferences::FlotsamCollection, 1> Preferences::flotsamCollection(
	{"off", "on", "flagship only", "escorts only"});
```